src/app/features/tech-record/components/tyres-search/tyres-search.component.ts

```diff
@@ -0,0 +1,171 @@
+import { Roles } from '@models/roles.enum';
+import { ReferenceDataService } from '@services/reference-data/reference-data.service';
+import { Router, ActivatedRoute } from '@angular/router';
+import { fetchReferenceDataByKeySearchSuccess, fetchTyreReferenceDataByKeySearchSuccess, ReferenceDataState } from '@store/reference-data';
+import { Store } from '@ngrx/store';
+import { selectTyreSearchReturn } from '@store/reference-data/selectors/reference-data.selectors';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
+import { updateEditingTechRecord } from '@store/technical-records/actions/technical-record-service.actions';
+import { VehicleTechRecordModel, TechRecordModel } from '@models/vehicle-tech-record.model';
+import { Actions, ofType } from '@ngrx/effects';
+
+@Component({
+  selector: 'app-tyres-search',
+  templateUrl: './tyres-search.component.html',
+  styleUrls: ['./tyres-search.component.scss']
+})
+export class TyresSearchComponent implements OnInit {
+  options?: MultiOptions = [
+    { label: 'Tyre code', value: 'code' },
+    { label: 'Ply rating', value: 'plyrating' },
+    { label: 'Single load index', value: 'singleload' },
+    { label: 'Double load index', value: 'doubleload' }
+  ];
+
+  constructor(
+    private cdr: ChangeDetectorRef,
+    public dfs: DynamicFormService,
+    public globalErrorService: GlobalErrorService,
+    private referenceDataService: ReferenceDataService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private technicalRecordService: TechnicalRecordService,
+    private store: Store<TechnicalRecordServiceState>,
+    private actions$: Actions
+  ) {
+    this.technicalRecordService.selectedVehicleTechRecord$.subscribe(data => (this.vehicleTechRecord = data));
+  }
+
+  public form!: CustomFormGroup;
+  public searchResults: Array<ReferenceDataTyre> | null = null;
+  public vehicleTechRecord?: VehicleTechRecordModel;
+  public viewableTechRecord: TechRecordModel | undefined = undefined;
+  private params: Params = {};
+
+  public template: FormNode = {
+    name: 'criteria',
+    type: FormNodeTypes.GROUP,
+    children: [
+      {
+        name: 'filter',
+        label: 'Search filter',
+        value: '',
+        type: FormNodeTypes.CONTROL
+      },
+      {
+        name: 'term',
+        value: '',
+        type: FormNodeTypes.CONTROL
+      }
+    ]
+  };
+
+  ngOnInit() {
+    this.form = this.dfs.createForm(this.template) as CustomFormGroup;
+    this.globalErrorService.clearErrors();
+    this.route.params.subscribe(p => (this.params = p));
+    this.technicalRecordService.editableTechRecord$.pipe().subscribe(data => (this.viewableTechRecord = data));
+    this.referenceDataService
+      .getTyreSearchReturn$()
+      .pipe(take(1))
+      .subscribe((data: any) => {
+        this.searchResults = data;
```

Comment:
why would you have tyre search result on load? are they not retrieved after actively searching by clicking on the button?

Reply:
The pagination causes the page to reload on init, so therefore you lose the data, or is there a better way around this?